app/src/main/kotlin/com/simplemobiletools/calculator/activities/MoneyActivity.kt

```diff
@@ -42,22 +55,134 @@ class MoneyActivity : SimpleActivity(), Calculator , TaxCalculator {
         updateViewColors(money_holder, config.textColor)
         updateButtonColor(config.customPrimaryColor)
 
+        locationManager = getSystemService(Context.LOCATION_SERVICE) as LocationManager
+
+        //Checks for permission granted for location service
+        if (ActivityCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_FINE_LOCATION)!= PackageManager.PERMISSION_GRANTED) {
+            //if not granted then request permission
+            ActivityCompat.requestPermissions(this, arrayOf(android.Manifest.permission.ACCESS_FINE_LOCATION), GPS_REQUEST_CODE);
+        }
+        try {
+            locationManager!!.requestLocationUpdates(LocationManager.GPS_PROVIDER, MINIMUM_TIME_BETWEEN_UPDATES, MINIMUM_DISTANCE_CHANGE_FOR_UPDATES, this)
+        } catch (e: SecurityException) {
+            print("NO PERMISSION GRANTED!")
+        }
+
         getButtonIds().forEach {
             it.setOnClickListener { calc.numpadClicked(it.id); checkHaptic(it) }
         }
 
         btn_currency.setOnClickListener{ true } // TODO : Implement feature and connect
         btn_delete.setOnClickListener { calc.handleDelete(); checkHaptic(it) }
         btn_delete.setOnLongClickListener { calc.handleClear(); true }
-        btn_taxes.setOnClickListener{ calc.calculateTax() } // TODO : Implement feature and connect
         btn_tip.setOnClickListener { true } // TODO : Implement feature and connect
         result.setOnLongClickListener { copyToClipboard(result.value); true }
+        btn_taxes.setOnClickListener({ view -> taxLocationStrat() })
 
         AutofitHelper.create(result)
 
     }
 
-     override fun spawnTaxModal() {
+    override fun onLocationChanged(location: Location) {
```

Comment:
Is this method every called? Also consider making this method private, we shouldn't be exposing this functionality.

Reply:
This method is called whenever a new location is being updated by line 66. This method is also being implemented from the interface LocationListener therefore can not be private